usaspending_api/download/management/commands/generate_spark_download.py

```diff
@@ -0,0 +1,224 @@
+        download_source = DownloadSource(
+            VALUE_MAPPINGS[self.download_type]["table_name"],
+            self.download_level,
+            self.download_type,
+            json_request.get("agency", "all"),
+            # TODO: Is this necessary for Spark downloads? It was originally added to File C downloads for performance.
+            extra_file_type="",
+        )
+        download_source.file_name = build_data_file_name(download_source, download_job, piid=None, assistance_id=None)
+
+        return download_job, download_source
+
+    def process_download(self):
+        self.start_download()
+        files_to_cleanup = []
+        try:
+            spark_to_csv_strategy = SparkToCSVStrategy(self.logger)
+            zip_file_path = self.working_dir_path / f"{self.download_name}.zip"
+            download_request = json.loads(self.download_job.json_request)
+            account_download_filter = AccountDownloadFilter(
+                year=int(download_request["filters"]["fy"]),
+                month=int(download_request["filters"]["period"]) if "period" in download_request["filters"] else None,
+                quarter=(
+                    int(download_request["filters"]["quarter"]) if "quarter" in download_request["filters"] else None
+                ),
+                agency=download_request["filters"].get("agency"),
+                federal_account_id=download_request["filters"].get("federal_account"),
+                def_codes=download_request["filters"].get("def_codes"),
+            )
+            source_df = self.df_builder(spark=self.spark, account_download_filter=account_download_filter).source_df
+            csv_metadata = spark_to_csv_strategy.download_to_csv(
+                source_sql=None,
+                destination_path=self.working_dir_path / self.download_name,
+                destination_file_name=self.download_name,
+                working_dir_path=self.working_dir_path,
+                download_zip_path=zip_file_path,
+                source_df=source_df,
+            )
+            files_to_cleanup.extend(csv_metadata.filepaths)
+            self.download_job.file_size = os.stat(zip_file_path).st_size
+            self.download_job.number_of_rows = csv_metadata.number_of_rows
+            self.download_job.number_of_columns = csv_metadata.number_of_columns
+            upload_download_file_to_s3(zip_file_path)
+        except InvalidParameterException as e:
+            exc_msg = "InvalidParameterException was raised while attempting to process the DownloadJob"
+            self.fail_download(exc_msg, e)
+            raise
+        except Exception as e:
+            exc_msg = "An exception was raised while attempting to process the DownloadJob"
+            self.fail_download(exc_msg, e)
+            raise
+        finally:
+            if self.should_cleanup:
+                self.cleanup(files_to_cleanup)
+        self.finish_download()
+
+    def start_download(self) -> None:
```

Comment:
Not something worth blocking for this work PR, but calling it out for future work. We need to look at implementing tracing with this management command like we have for the rest of the downloads so that it is easier to debug once this is in production.